tests/api_tests/comments/test_views.py

```diff
@@ -0,0 +1,889 @@
+        assert_equal(res.status_code, 409)
+
+    def test_report_comment_no_type(self):
+        payload = {
+            'data': {
+                'type': '',
+                'attributes': {
+                    'category': 'spam',
+                    'message': 'delicious spam'
+                }
+            }
+        }
+        res = self.app.post_json_api(self.private_url, payload, auth=self.user.auth, expect_errors=True)
+        assert_equal(res.status_code, 400)
+        assert_equal(res.json['errors'][0]['detail'], 'This field may not be blank.')
+        assert_equal(res.json['errors'][0]['source']['pointer'], '/data/type')
+
+    def test_report_comment_invalid_spam_category(self):
+        payload = {
+            'data': {
+                'type': 'comment_reports',
+                'attributes': {
+                    'category': 'Not a valid category',
+                    'message': 'delicious spam'
+                }
+            }
+        }
+
+        res = self.app.post_json_api(self.private_url, payload, auth=self.user.auth, expect_errors=True)
+        assert_equal(res.status_code, 400)
+
+    def test_report_comment_allow_blank_message(self):
+        comment = CommentFactory(node=self.private_project, user=self.contributor)
+        url = '/{}comments/{}/reports/'.format(API_BASE, comment._id)
+        payload = {
+            'data': {
+                'type': 'comment_reports',
+                'attributes': {
+                    'category': 'spam',
+                    'message': ''
+                }
+            }
+        }
+        res = self.app.post_json_api(url, payload, auth=self.user.auth)
+        assert_equal(res.status_code, 201)
+        assert_equal(res.json['data']['id'], self.user._id)
+        assert_equal(res.json['data']['attributes']['message'], payload['data']['attributes']['message'])
+
+    def test_private_node_logged_out_user_cannot_report_comment(self):
+        res = self.app.post_json_api(self.private_url, self.payload, expect_errors=True)
+        assert_equal(res.status_code, 401)
+
+    def test_private_node_logged_in_non_contributor_cannot_report_comment(self):
+        res = self.app.post_json_api(self.private_url, self.payload, auth=self.non_contributor.auth, expect_errors=True)
+        assert_equal(res.status_code, 403)
+
+    def test_private_node_logged_in_contributor_can_report_comment(self):
+        comment = CommentFactory(node=self.private_project, user=self.contributor)
+        url = '/{}comments/{}/reports/'.format(API_BASE, comment._id)
+
+        res = self.app.post_json_api(url, self.payload, auth=self.user.auth)
+        assert_equal(res.status_code, 201)
+        assert_equal(res.json['data']['id'], self.user._id)
+
+    def test_user_cannot_report_own_comment(self):
+        res = self.app.post_json_api(self.private_url, self.payload, auth=self.contributor.auth, expect_errors=True)
+        assert_equal(res.status_code, 400)
```

Comment:
Since 400 is a fairly generic error code, try to also check the text of the error message when possible. (to be sure that the test is failing for the reason you expect)